kstreams/types.py

```diff
@@ -1,9 +1,13 @@
 import typing
+from contextlib import AbstractAsyncContextManager
 from dataclasses import dataclass
 
 from aiokafka.structs import RecordMetadata
 
+from .clients import Producer
+
 if typing.TYPE_CHECKING:
+    from . import transaction  #  pragma: no cover
```

Comment:
is there a way to avoid this circular import? could `transaction.Transaction` also be a protocol?